yb-voyager/cmd/exportData_test.go

```diff
@@ -659,3 +749,330 @@ func testCasesWithDifferentTableListFlagValuesTest1(t *testing.T, firstRunTableL
+	}
+
+	metaDB = initMetaDB(testExportDir)
+
+	//Running the command level functions
+	source = *testPostgresSource.Source
+
+	//Fetch table list and partitions to root mapping in the first run
+	expectedPartitionsToRootMapWithoutFlags := map[string]string{
+		`test1.cust_part22`: `test1.customers`,
+		`test1.cust_part21`: `test1.customers`,
+		`test1.cust_part12`: `test1.customers`,
+		`test1.cust_part11`: `test1.customers`,
+		`test2.cust_other`:  `test1.customers`,
+	}
+
+	expectedTableListWithoutFlags := []sqlname.NameTuple{
+		getNameTuple(`test1.cust_part22`),
+		getNameTuple(`test1.cust_part21`),
+		getNameTuple(`test1.cust_part12`),
+		getNameTuple(`test1.cust_part11`),
+		getNameTuple(`test2.cust_other`),
+		getNameTuple(`test1.customers`),
+		getNameTuple(`test1."Foo"`),
+		getNameTuple(`test1."Foo1"`),
+		getNameTuple(`test1."Foo2"`),
+	}
+
+	assertTableListFilteringInTheFirstRun(t, expectedPartitionsToRootMapWithoutFlags, expectedTableListWithoutFlags)
+
+	//case table-list
+	//--table-list test1.customers,test1.Foo* --exclude-table-list test2.cust_arr*
+	source.TableList = "test1.customers,test1.Foo*"
+	source.ExcludeTableList = "test2.cust_arr*"
+
+	//Fetch table list and partitions to root mapping in the first run
+	expectedPartitionsToRootMap := map[string]string{
+		`test2.cust_other`: `test1.customers`,
+	}
+
+	expectedTableList := []sqlname.NameTuple{
+		getNameTuple(`test2.cust_other`),
+		getNameTuple(`test1.customers`),
+		getNameTuple(`test1."Foo"`),
+		getNameTuple(`test1."Foo1"`),
+		getNameTuple(`test1."Foo2"`),
+	}
+
+	assertTableListFilteringInTheFirstRun(t, expectedPartitionsToRootMap, expectedTableList)
+
+	expectedTableListWithOnlyRootTable := []sqlname.NameTuple{}
+
+	//Create msr with required details for subsequent run
+	err = metaDB.UpdateMigrationStatusRecord(func(msr *metadb.MigrationStatusRecord) {
+		msr.SourceDBConf = testPostgresSource.Source
+		msr.TableListExportedFromSource = lo.Map(expectedTableListWithOnlyRootTable, func(t sqlname.NameTuple, _ int) string {
+			return t.ForOutput()
+		})
+		msr.SourceExportedTableListWithLeafPartitions = lo.Map(expectedTableList, func(t sqlname.NameTuple, _ int) string {
+			return t.ForOutput()
+		})
+		msr.SourceRenameTablesMap = expectedPartitionsToRootMap
+	})
+	if err != nil {
+		t.Fatalf("error updating msr: %v", err)
+	}
+
+	testCasesWithDifferentTableListFlagValuesTest2(t, expectedTableList, expectedPartitionsToRootMap)
+
+}
+
+func testCasesWithDifferentTableListFlagValuesTest2(t *testing.T, firstRunTableList []sqlname.NameTuple, firstRunPartitionsToRootMap map[string]string) {
+	//case1: getInitialTableList for subsequent run with  start-clean false and basic with same table-list flags so no guardrails
+	assertInitialTableListOnSubsequentRun(t, false, firstRunTableList, firstRunPartitionsToRootMap)
+
+	//case2: getInitialTableList for subsequent run with  start-clean false and basic with no table-list flags so reporting extra tables found
```

Comment:
if we provide NO table-list flags, shouldn't we report "missing" tables, rather than "extra" tables?

Reply:
In case no table-list flags are used we consider the full table list as the input so we use the name-registry tablelist in that case and then report the guardrails accordingly

Reply:
Ah, I see. 
first run: table-list provided. 
subsequent run: table-list NOT provided. (i.e. consider all tables)

Therefore, in the subsequent run, we report "extra" tables. Correct?

Reply:
Right